skbio/metadata/_interval.py

```diff
@@ -0,0 +1,599 @@
+
+        Returns
+        -------
+        bool
+            Indicates if the two objects are not equal.
+        '''
+        return not self.__eq__(other)
+
+    @experimental(as_of='0.4.2-dev')
+    def __repr__(self):
+        '''Return a string representation of this ``Interval`` object.
+
+        Returns
+        -------
+        str
+            String representation of this ``Interval`` object.
+        '''
+        s = '{}(interval_metadata=<{!r}>, locations={!r}, boundaries={!r}, metadata={!r})'
+        return s.format(self.__class__.__name__, id(self._interval_metadata),
+                        self.locations, self.boundaries, self.metadata)
+
+    @experimental(as_of='0.4.2-dev')
+    def drop(self):
+        '''Drop this ``Interval`` object from the interval metadata it links to.
+
+        See Also
+        --------
+        skbio.metadata.IntervalMetadata.drop
+        '''
+        self._interval_metadata.drop(locations=self.locations,
+                                     metadata=self.metadata)
+        self._interval_metadata = None
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def boundaries(self):
+        '''The openness of each coordinate.
+
+        This indicates that the exact boundary point of a interval feature
+        is unknown. The location may begin or end at some points outside
+        the specified coordinates. This accommodates the location format [1]_
+        of INSDC.
+
+        References
+        ----------
+        .. [1] ftp://ftp.ebi.ac.uk/pub/databases/embl/doc/FT_current.html#3.4.3
+        '''
+        if self.dropped:
+            raise RuntimeError('Cannot retrieve boundaries from.'
+                               'dropped Interval object.')
+        return self._boundaries
+
+    @boundaries.setter
+    @experimental(as_of='0.4.2-dev')
+    def boundaries(self, value):
+        if value is None:
+            self._boundaries = None
+        else:
+            for boundary in value:
+                _assert_valid_boundary(boundary)
+
+            if self.dropped:
+                raise RuntimeError('Cannot change boundaries to dropped '
+                                   'Interval object.')
+            self._boundaries = value
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def locations(self):
+        '''The coordinates of the interval feature.
+
+        It should be a list of tuples of int pair. Each tuple stores
+        the start and end coordinates of a span of the interval feature.
+        '''
+        if self.dropped:
+            raise RuntimeError(
+                'Cannot retrieve locations from dropped Interval object.')
+        return self._locations
+
+    @locations.setter
+    @experimental(as_of='0.4.2-dev')
+    def locations(self, value):
+        if value is None:
+            self._locations = None
+        else:
+            for location in value:
+                _assert_valid_location(location)
+
+            if self.dropped:
```

Comment:
should this check be performed prior to `_assert_valid_location` as, if dropped, it doesn't matter if the location is valid or not? would it make sense prior to checking `if value is None` or is it possible to have location information even if `dropped is True`?